lib/Mojo/Server/CGI.pm

```diff
@@ -3,6 +3,14 @@ use Mojo::Base 'Mojo::Server';
 
 has 'nph';
 
+sub new {
+  my $class = shift;
+  my $self  = $class->SUPER::new(@_);
+  my $app   = $self->app;
+  $app->worker_starting if ref $app;
+  return $self;
+}
```

Comment:
This is now inconsistent with `before_server_start`. Also a blocker for me.

Reply:
How is it inconsistent exactly?

Reply:
`Mojo::Server::CGI` is excluded from deployment hooks like `before_server_start`.

Reply:
And this is also a deployment hook you are working on. It does not get emitted for commands like `./myapp.pl eval -V 'app->mode'` after all.

Reply:
@kraih you are right, something is not working as expected yet.  I only discovered Mojo 2 days ago, hence I apologize for being a slow learner.
I really wish to have a global hook that triggers possibly costly initialization for the application, only **once** and regardless how the application is actually started.

Reply:
@krait There, now it runs exactly once per process.
It will get run if you do `./myapp.pl get /` but not if you do `./myapp.pl eval -V ...` because then you are not going to serve a request, hence why perform global initialization only required for request serving?

Reply:
And that makes it inconsistent with `before_server_start`. Which does not run for `./myapp.pl cgi`.

Reply:
I think this is a blocker too. We might have to consider adding the `before_server_start` hook to cgi first.

Reply:
Why do you try to link the fate of `before_server_start` with that of `before_app_start`? They have different semantics now.

We could add `before_server_start` to CGI, but that would be another PR, because, for me, the two events have different semantics, even if, at first, when I started toying with this idea, I emitted one close to the other...

Reply:
Yes, i think `before_server_start` should be added to `Mojo::Server::CGI` after all.

Reply:
And done. https://github.com/mojolicious/mojo/compare/81a2753e992e...8e1611e3682a